pypeit/core/procimg.py

```diff
@@ -328,61 +374,82 @@ def rect_slice_with_mask(image, mask, mask_val=1):
+            set ``params`` to the order, number of pixels, number of repeats;
+            for ``method=savgol``, set ``params`` to the order and window size;
+            for ``method=median``, ``params`` are ignored.
+        var (`numpy.ndarray`_, optional):
+            Variance in the raw frame.  If None, ignored.  If provided, an
+            estimate of the error in the overscan correction is included and an
+            updated variance image is returned.  The estimated error is the
+            standard error in the median for the pixels included in the overscan
+            correction.  This estimate is also used for the ``'savgol'`` method
+            as an upper limit.
 
     Returns:
-        :obj:`numpy.ndarray`: The input frame with the overscan region
-        subtracted
+        `numpy.ndarray`_, :obj:`tuple`: The input frame with the overscan region
+        subtracted.  If ``var`` is provided, the returned object is a tuple with
+        an estimate of the propagated variance in the overscan-subtracted image
+        as the second item.
     """
+    # Check input
+    if method.lower() not in ['polynomial', 'savgol', 'median']:
+        msgs.error(f'Unrecognized overscan subtraction method: {method}')
+
     # Copy the data so that the subtraction is not done in place
     no_overscan = rawframe.copy()
+    _var = None if var is None else var.copy()
 
     # Amplifiers
     amps = np.unique(datasec_img[datasec_img > 0]).tolist()
 
     # Perform the overscan subtraction for each amplifier
     for amp in amps:
         # Pull out the overscan data
-        overscan, _ = rect_slice_with_mask(rawframe, oscansec_img, amp)
+        if np.sum(oscansec_img == amp) == 0:
+            msgs.error(f'No overscan region for amplifier {amp+1}!')
+        overscan, os_slice = rect_slice_with_mask(rawframe, oscansec_img, amp)
+        if var is not None:
+            osvar = var[os_slice]
         # Pull out the real data
+        if np.sum(datasec_img == amp) == 0:
+            msgs.error(f'No data region for amplifier {amp+1}!')
         data, data_slice = rect_slice_with_mask(rawframe, datasec_img, amp)
 
         # Shape along at least one axis must match
-        data_shape = data.shape
-        if not np.any([dd == do for dd, do in zip(data_shape, overscan.shape)]):
+        if not np.any([dd == do for dd, do in zip(data.shape, overscan.shape)]):
             msgs.error('Overscan sections do not match amplifier sections for'
                        'amplifier {0}'.format(amp))
-        compress_axis = 1 if data_shape[0] == overscan.shape[0] else 0
+        compress_axis = 1 if data.shape[0] == overscan.shape[0] else 0
 
         # Fit/Model the overscan region
         osfit = np.median(overscan) if method.lower() == 'median' \
-            else np.median(overscan, axis=compress_axis)
+                    else np.median(overscan, axis=compress_axis)
+        if var is not None:
```

Comment:
I would have thought this error was pretty small, but I see no harm in including it.